pkg/gateway/client/user.go

```diff
@@ -98,16 +120,40 @@ func (c *Client) DeleteUser(ctx context.Context, userID string) (*types.User, er
 			}
 		}
 
-		if err := tx.Where("user_id = ?", existingUser.ID).Delete(new(types.Identity)).Error; err != nil {
+		// Soft delete: set timestamp and preserve original values
+		now := time.Now()
+		existingUser.DeletedAt = &now
+		existingUser.OriginalEmail = existingUser.Email
+		existingUser.OriginalUsername = existingUser.Username
+
+		// Clear sensitive data and make fields unique to allow re-signup
+		// Append timestamp to prevent conflicts with new users using same email
+		deletedSuffix := fmt.Sprintf("_deleted_%d", now.Unix())
+		existingUser.Email = existingUser.Email + deletedSuffix
+		existingUser.Username = existingUser.Username + deletedSuffix
+		existingUser.HashedEmail = hash.String(existingUser.Email)
+		existingUser.HashedUsername = hash.String(existingUser.Username)
+
+		// Encrypt the modified user
+		if err := c.encryptUser(ctx, existingUser); err != nil {
+			return fmt.Errorf("failed to encrypt user: %w", err)
+		}
+
+		// Update the user with soft delete fields and modified email/username
+		if err := tx.Save(existingUser).Error; err != nil {
 			return err
 		}
 
-		return tx.Delete(existingUser).Error
+		return nil
 	}); err != nil {
 		return nil, err
 	}
 
-	return existingUser, c.decryptUser(ctx, existingUser)
+	// Return the user with original values for the response (before encryption)
+	responseUser := *existingUser
+	responseUser.Email = responseUser.OriginalEmail
+	responseUser.Username = responseUser.OriginalUsername
+	return &responseUser, nil
```

Comment:
The intention is that the encrypted values for a user never leave this method. Since the `existingUser` is being encrypted on line 138 above, we should decrypt it here before returning. There have been a few bugs in this area caused by yours truly.

Instead of decrypting here, you could copy `existingUser` and copy over the values like you are doing here before encrypting `existingUser` above. Then here, just `return &responseUser, nil`.